### PR TITLE
[webgpu] Simplify activation shader key

### PR DESCRIPTION
**Pull request by axinging**
1. Use backend_util.Activation as activation key instead of activation string. This make shader key shorter, save warmup time.
2. Move shader related code to program, this decouples the general ts code and shader code.
3. For future WGSL, the activation(glsl or wgsl) can be decided by each program.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5208)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @haoyunfeix @xhcao @gyagp , PTAL

File: tfjs-backend-webgpu/src/kernels/conv2d_mm_vec4_webgpu.ts [hunk 1]
```diff
@@ -72,8 +73,8 @@ export class Conv2DMMVec4Program implements WebGPUProgram {
     }
 
     [this.fitA, this.fitB] = this.getShapeFit(elementsPerThread);
-    this.shaderKey =
-        `conv2DMMVec4_${this.activation}_${this.fitA}_${this.fitB}`;
+    this.shaderKey = `conv2DMMVec4_${this.activation}_${this.isVec4}_${
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/conv2d_mm_vec4_webgpu.ts:76]: `this.isVec4` is not needed. `conv2DMMVec4` already means it's always true.

File: tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts [hunk 2]
```diff
@@ -211,8 +213,8 @@ export class MatMulPackedVec4Program implements WebGPUProgram {
 
     [this.fitA, this.fitB] = this.getShapeFit();
 
-    this.shaderKey = `matMulPackedVec4_${rowPerThread}_${activation}_${
-        this.fitA}_${this.fitB}_${this.outputShape[1] > 1}`;
+    this.shaderKey = `matMulPackedVec4_${rowPerThread}_${this.activation}_${
+        this.isVec4}_${this.fitA}_${this.fitB}_${this.outputShape[1] > 1}`;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts:217]: Same here.

**Review by qjia7 — APPROVED** [review]
LGTM with nits.

**Comment by qjia7** [comment]
@lina128 @jinjingforever @kainino0x take a look, thanks.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5208#-:-Mbw8iFh-AkokwcwO1EC:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @axinging, @jinjingforever, and @kainino0x)

**Review by kainino0x — APPROVED** [review]

File: tfjs-backend-webgpu/src/kernels/conv2d_mm_vec4_webgpu.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/conv2d_mm_vec4_webgpu.ts]: done

File: tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts (hunk 2 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/matmul_packed_vec4_webgpu.ts]: done

**Pull request merged**